authentication/java/azure/lib/src/main/java/com/microsoft/kiota/authentication/AzureIdentityAccessTokenProvider.java

```diff
@@ -33,9 +35,20 @@ public AzureIdentityAccessTokenProvider(@Nonnull final TokenCredential tokenCred
         } else {
             _scopes = Arrays.asList(scopes);
         }
+        if (allowedHosts == null || allowedHosts.length == 0) {
+            _hostValidator = new AllowedHostsValidator(new String[] { "graph.microsoft.com", "graph.microsoft.us", "dod-graph.microsoft.us", "graph.microsoft.de", "microsoftgraph.chinacloudapi.cn", "canary.graph.microsoft.com" });
```

Comment:
default valid hostnames values and default scope value in the access token provider should be defined in graph core

Reply:
created https://github.com/microsoft/kiota/issues/1063 to solve that in a later pull request

Reply:
Should #1063 also include work to allow the passing of an empty Array to the AllowedHostsValidator?
I say this because the `AllowedHostsValidator` will also allow all hosts if we pass an empty array and the `allowedHosts.length == 0` branch currently prevents the realization of that scenario when using this tokenProvider

Reply:
This applies to other languages as well.

Reply:
null or empty array means all requests get a token, valued array means only requests which hostname matches one of the entries get a token. Hopefully this clarifies the intent.

Reply:
that's for the allowed host validator.
For the access token provider, we're defaulting the value to the graph values for the time being should an empty or null array be passed. And we'll move those defaults into the graph core libs with #1063